ddtrace/internal/runtime/container.py

```diff
@@ -14,15 +13,15 @@
 log = get_logger(__name__)
 
 
-@attr.s(slots=True)
-class CGroupInfo(object):
-    id = attr.ib(default=None)
-    groups = attr.ib(default=None)
-    path = attr.ib(default=None)
-    container_id = attr.ib(default=None)
-    controllers = attr.ib(default=None)
-    pod_id = attr.ib(default=None)
-    node_inode = attr.ib(default=None)
+class CGroupInfo:
```

Comment:
any reason to not use a dataclass here? especially since you want an __eq__?

Reply:
We can use `dataclass` and set `__slots__` manually, as far as we don't set default values for the attributes. I can try that, and if it works, add a warning to avoid setting defaults in the future?

Reply:
> as far as we don't set default values for the attributes

oh... 🤔 we cannot do slots and default values?

Reply:
```python
Python 3.10.13 (main, Dec 14 2023, 17:54:16) [Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from dataclasses import dataclass
>>>
>>> @dataclass
... class C():
...     __slots__ = "x"
...     x: int = 1
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: 'x' in __slots__ conflicts with class variable
```